siliconcompiler/core.py

```diff
@@ -2771,9 +2773,104 @@ def _find_leaves(self, steplist):
         # Then, find all leaf tasks by elimination.
         return all_tasks.difference(non_leaf_tasks)
 
+    ###########################################################################
+    def _generate_summary_image(self, input_path, output_path):
+        '''Takes a layout screenshot and generates a design summary image
+        featuring a layout thumbnail and several metrics.'''
+
+        # Extract metrics for display
+
+        metrics = {
+            'Chip': self.get('design'),
+        }
+
+        if self.get('option', 'pdk'):
+            metrics['Node'] = self.get('option', 'pdk')
+
+        # TODO: a bit hardcoded to asicflow assumptions... a way to query
+        # "final" metrics regardless of flow would be handy
+        totalarea = self.get('metric', 'totalarea', step='export', index='1')
+        if totalarea:
+            metrics['Area'] = f'{totalarea:.2f} um^2'
+
+        fmax = self.get('metric', 'fmax', step='export', index='1')
+        if fmax:
+            # SI-ify
+            if fmax < 1e6:
+                fmax = f'{(fmax / 1e3):.2f} KHz'
+            elif fmax < 1e9:
+                fmax = f'{(fmax / 1e6):.2f} MHz'
+            else:
+                fmax = f'{(fmax / 1e9):.2f} GHz'
+            metrics['Fmax'] = fmax
+
+        # Generate design
+
+        WIDTH = 512
```

Comment:
Could we go with a higher-resolution? The 4K screenshots I looked at were only be in the range of 4-5MB before gzipping, but 1024 or 2048 could also provide better detail on large designs.

Reply:
Fair point, scaled it up to 1024.